### DISCUSSION TITLE
Help: Missing countries - Volunteers needed

### DISCUSSION THREAD

**Discussion by derTobsch** [focus-shift/jollyday#522] [General]
There are actually these missing countries:
- [ ] Asia 
  - [ ] Afghanistan
  - [ ] Brunei
  - [ ] Bahrain
  - [ ] Afghanistan
  - [ ] Armenia
  - [ ] Azerbaijan
  - [ ] Bahrain
  - [ ] Bangladesh
  - [ ] Bhutan
  - [ ] Brunei
  - [ ] Cambodia
  - [ ] Indonesia
  - [ ] Iran
  - [ ] Iraq
  - [ ] Israel
  - [ ] Jordan
  - [ ] Kuwait
  - [ ] Kyrgyzstan
  - [ ] Laos
  - [ ] Lebanon
  - [ ] Malaysia
  - [ ] Maldives
  - [ ] Mongolia
  - [ ] Nepal
  - [ ] Oman
  - [ ] Pakistan
  - [ ] Philippines
  - [ ] Qatar
  - [ ] Sri Lanka
  - [ ] Syria
  - [ ] Tajikistan
  - [ ] Thailand
  - [ ] Timor-Leste
  - [ ] Turkmenistan
  - [ ] Uzbekistan
  - [ ] Yemen
- [ ] Africa 
  - [ ] Burundi
  - [ ] Cape Verde
  - [ ] Comoros
  - [ ] Djibouti
  - [ ] Equatorial Guinea
  - [ ] Eritrea
  - [ ] Guinea Bissau
  - [ ] Lesotho
  - [ ] Liberia
  - [ ] Mozambique
  - [ ] Niger
  - [ ] Rwanda
  - [ ] Seychelles
  - [ ] Somalia
- [ ] North America
  - [ ] Barbados
  - [ ] Dominica
  - [ ] Dominican Republic
  - [ ] Grenada
- [ ] South America
  - [ ] Guyana
  - [ ] Suriname
- [ ] Oceania
  - [ ] Cook Islands
  - [ ] Fiji
  - [ ] Micronesia
  - [ ] Niue
  - [ ] Samoa
  - [ ] Tonga
  - [ ] Vanuatu
  - [ ] Papua New Guinea

If anyone is interested in providing the public holidays for one of these countries, they are welcome to join this discussion.

**Comment by CodeTectonics** [comment]
Brunei's public holidays (with dates from 2024)

- 1st January - New Year's Day
- 7th February - Israk Mikraj
- 10th February - Lunar New Year
- 23rd February - National Day
- 11th March - First Day Of Ramadhan
- 28th March - Nuzul Al-Quran
- 10th April - Hari Raya Aidil Fitri
- 11th April - Hari Raya Aidil Fitri Holiday
- 12th April - Hari Raya Aidil Fitri Holiday
- 31st May - Royal Brunei Armed Forces Day
- 17th June - Hari Raya Aidil Adha
- 8th July - Al-Hijra
- 15th July - Sultan's Birthday
- 15th September - Prophet Muhammad's Birthday
- 25th December - Christmas Day

Notes:
 * The dates of the following holidays are based on a lunar calendar, so they will change every year, but will be standard with several other countries:
	 * Israk Mikraj
	 * Lunar New Year
	 * First Day Of Ramadhan
	 * Nuzul Al-Quran
	 * Hari Raya Aidil Adha
	 * Al-Hijra
	 * Prophet Muhammad's Birthday
 * The dates of the following holidays are based on a combination of the lunar calendar and a manual moon-sighting by a government officer, so may (and often do) differ from other countries with the same holidays:
	 * Hari Raya Aidil Fitri
	 * Hari Raya Aidil Fitri Holiday
	 * Hari Raya Aidil Fitri Holiday

**Reply by derTobsch** [reply]
Hey @CodeTectonics,

thanks for your effort. What we need to add countries to Jollyday, are the "rules" that are behind every public holiday as you can see for morocco here https://github.com/focus-shift/jollyday/pull/570/files#diff-242cb023a5d40ee71889c9954e9f8a8e3fa6462407febf04bdf609231c9dc0d3

There are different types of holidays and we need the "rule" how they will be calculated. Are you familiar with the rules behind the public holidays of Brunei?

**Comment by fauzaanu** [comment]
Hello, I Could volunteer to do Maldives. 

Not very clear on the process though, but for Maldives [this](https://en.wikipedia.org/wiki/Public_holidays_in_the_Maldives) is a good starting point

**Reply by derTobsch** [reply]
Hey @fauzaanu,
thanks you that you want to volunteer. A good starting point may be the pull request for the Morocco public holidays see here https://github.com/focus-shift/jollyday/pull/570/files#diff-242cb023a5d40ee71889c9954e9f8a8e3fa6462407febf04bdf609231c9dc0d3

We need to define the rules behind the holidays.
